eth/polls/eth_subscribe.go

```diff
@@ -26,60 +27,79 @@ type EthPoll interface {
 
 type EthSubscriptions struct {
 	polls map[string]EthPoll
-
-	lastPoll    map[string]uint64
-	timestamps  map[uint64][]string
-	lastPrune   uint64
-	evmAuxStore *evmaux.EvmAuxStore
-	blockStore  store.BlockStore
+	lastPoll        map[string]uint64
+	timestamps      map[uint64][]string
+	mutex      *sync.RWMutex
```

Comment:
This shouldn't be a pointer, that way you'll get a nice warning if you try to do something silly like copying the struct it's embedded in.

Reply:
Holding mutex by value generates a lint error [passes lock by value](https://medium.com/golangspec/detect-locks-passed-by-value-in-go-efb4ac9a3f2b)
```
[2019-07-19T12:21:17.260Z] rpc/query_server.go:276:9: EthCall passes lock by value: github.com/loomnetwork/loomchain/rpc.QueryServer contains github.com/loomnetwork/loomchain/eth/polls.EthSubscriptions contains sync.RWMutex (govet)
[2019-07-19T12:21:17.260Z] func (s QueryServer) EthCall(query eth.JsonTxCallObject, block eth.BlockHeight) (resp eth.Data, err error) {
[2019-07-19T12:21:17.260Z]         ^
[2019-07-19T12:21:17.260Z] rpc/query_server.go:707:9: GetEvmTransactionByHash passes lock by value: github.com/loomnetwork/loomchain/rpc.QueryServer contains github.com/loomnetwork/loomchain/eth/polls.EthSubscriptions contains sync.RWMutex (govet)
[2019-07-19T12:21:17.260Z] func (s QueryServer) GetEvmTransactionByHash(txHash []byte) (resp []byte, err error) {
[2019-07-19T12:21:17.260Z]         ^
[2019-07-19T12:21:17.260Z] rpc/query_server.go:958:9: EthNewBlockFilter passes lock by value: github.com/loomnetwork/loomchain/rpc.QueryServer contains github.com/loomnetwork/loomchain/eth/polls.EthSubscriptions contains sync.RWMutex (govet)
[2019-07-19T12:21:17.260Z] func (s QueryServer) EthNewBlockFilter() (eth.Quantity, error) {
[2019-07-19T12:21:17.260Z]         ^
[2019-07-19T12:21:17.260Z] rpc/query_server.go:965:9: EthNewPendingTransactionFilter passes lock by value: github.com/loomnetwork/loomchain/rpc.QueryServer contains github.com/loomnetwork/loomchain/eth/polls.EthSubscriptions contains sync.RWMutex (govet)
[2019-07-19T12:21:17.260Z] func (s QueryServer) EthNewPendingTransactionFilter() (eth.Quantity, error) {
```

Reply:
Yeah, the linter is doing it's job. Those functions you see the warning for should be changed `(s QueryServer) -> `(s *QueryServer)`